### PR TITLE
Do not parse invalid rev-parse responses

### LINKED ISSUE #6512: Exception if opening a submodule with incorrect superproject reference
Unable to parse object ID "fatal: not a git repository: 

## Current behaviour

If a repo to be opened is a submodule, and the superproject for some reason is corrupted (also if the submodule has been copied), opening the repo gives an exception: 

## Expected behaviour

No exception, showing that the repo has no commits (the normal behavior)

## Steps to reproduce

Get a submodule with incorrect link to the superrepo by modifying the superrepo or just copy the submodule:
$ cp -pr ../gitextensions/Externals/Git.hub/ .
Open the submodule and the error occurs

## Error Details
```
System.FormatException: Unable to parse object ID "fatal: not a git repository: F:/dev/gc/tmp/Git.hub/../../.git/modules/Externals/Git.hub".
   at GitUIPluginInterfaces.ObjectId.Parse(String s) in F:\dev\gc\gitextensions\Plugins\GitUIPluginInterfaces\ObjectId.cs:line 173
   at GitCommands.GitModule.GetCurrentCheckout() in F:\dev\gc\gitextensions\GitCommands\Git\GitModule.cs:line 1170
   at GitUI.RevisionGridControl.ForceRefreshRevisions() in F:\dev\gc\gitextensions\GitUI\UserControls\RevisionGrid\RevisionGridControl.cs:line 884
   at GitUI.CommandsDialogs.FormBrowse.RefreshRevisions() in F:\dev\gc\gitextensions\GitUI\CommandsDialogs\FormBrowse.cs:line 602
   at GitUI.GitUIExtensions.<InvokeAsync>d__11.MoveNext() in F:\dev\gc\gitextensions\GitUI\GitUIExtensions.cs:line 209
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at GitUI.ThreadHelper.<>c__DisplayClass13_0.<<FileAndForget>b__0>d.MoveNext() in F:\dev\gc\gitextensions\GitExtUtils\GitUI\ThreadHelper.cs:line 101
```

## Environment
- Git Extensions 3.1.0
- Build 33675bd986268ccbace666afe9bdbcbd34b012e8
- Git 2.21.0.windows.1
- Microsoft Windows NT 6.1.7601 Service Pack 1
- .NET Framework 4.7.3394.0
- DPI 96dpi (no scaling)

### PR DESCRIPTION
**Pull request by gerhardol**
Fixes #6512

## Proposed changes

Do not parse invalid rev-parse responses

## Test methodology 

Added a unit test
See issue for how to reproduce

## Test environment(s) 

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: UnitTests/GitCommandsTests/GitModuleTests.cs [hunk 1]
```diff
@@ -359,6 +359,15 @@ public void RevParse_should_query_git_and_return_null_if_invalid_response()
             }
         }
 
+        [Test]
+        public void GetCurrentCheckout_should_query_git_and_return_null_if_invalid_response()
+        {
+            using (_executable.StageOutput($"rev-parse HEAD", "fatal: not a git repository:", 0))
```
Comment (RussKie) [inline UnitTests/GitCommandsTests/GitModuleTests.cs]: Turn into a `TestCase` for both "fatal:" and "error:"

**Comment by RussKie** [comment]
> Now, it claims that the sub-repo with the missing parent repo had unresolved merge conflicts.

Any ideas how we can resolve this?

**Comment by RussKie** [comment]
Finally got to a normal computer.
Looking at the screenshot I suspect the initialisation routine wasn't allowed to finish completely that caused the UI to be presented in its default state.

We need to ensure we don't attempt to show the UI until we know we can open the repo. 
If we can't open the repo we should present the dialog to offer to remove the invalid repo
https://github.com/gitextensions/gitextensions/blob/cb425db6e505df8772e99235938cbc55252bab0d/GitUI/CommandsDialogs/FormBrowse.cs#L1669-L1681

**Comment by mstv** [comment]
> We need to ensure we don't attempt to show the UI until we know we can open the repo.

Yes, when the refactorings of #6348 will have been merged, especially `ISimpleDialog`, we should address Git errors more generally: A `GitException` should rather be thrown with the original Git error message as the `Exception.Message`. These exceptions should be caught by `GitModuleForm`/`GitModuleControl` and the error message be displayed. I recommend to wait for `ISimpleDialog` because it will make MessageBoxes testable better.

**Comment by gerhardol** [comment]
This is a minor fix that I found worthwhile to address after investigating the issue.
There are checks when opening repos already, but there will always be situations like this.
This check checks for fatal: and error: only. There can theoretically be other responses, we could exclude all with illegal characters:
https://git-scm.com/docs/git-check-ref-format

But this PR is a (minor)  improvement IMO

**Comment by RussKie** [comment]
Happy to take this into 3.1.

**Comment by gerhardol** [comment]
The 3.1.0 release branch is created, this can wait

> we should address Git errors more generally: A GitException should rather be thrown with the original Git error message

I aligned the handling to other git-rev-parse calls, will also simplify a "GitExtension" handling.
Added a positive testcase too.

Will squash if approved

**Pull request merged**